lang/mrustc/Portfile

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem      1.0
+PortGroup       github 1.0
+
+github.setup    thepowersgang mrustc aa4adf3504876f58c90a3b8770116fbefc7c2867
+
+version         20220103
+revision        0
+categories      lang devel
+license         MIT
+maintainers     nomaintainer
+description     Alternative implementation of Rust
+supported_archs x86_64
+
+long_description \
+                In-progress alternative rust compiler. Capable of building a \
+                fully-working copy of rustc, but not suitable for everyday use \
+                (due to inadequate error messages).
+
+master_sites-append   https://static.rust-lang.org/dist/
+
+set rust_version 1.39.0
+set rust_version_major [join [lrange [split ${rust_version} .-] 0 1] .]
+set rust_version_major_underscore [join [lrange [split ${rust_version} .-] 0 1] _]
+distfiles-append rustc-${rust_version}-src.tar.gz
+
+checksums       mrustc-${github.version}.tar.gz \
+                rmd160  1e76cd4451a096629c855c6786bbf5c93233872d \
+                sha256  4b58ffe5315081ea780512c8930875ae7d3fc41adfc4d7554c0738af50cbaaf6 \
+                size    1171492 \
+                rustc-${rust_version}-src.tar.gz \
+                rmd160  d5b04b87fc336e3be7d592f70de0363aa66622aa \
+                sha256  b4a1f6b6a93931f270691aba4fc85eee032fecda973e6b9c774cd06857609357 \
+                size    152803201
+
+use_configure   no
+
+compiler.cxx_standard 2014
+compiler.c_standard 2011
+
+# This will be the default compiler, which can be overridden
+# at run-time with the CC environment variable.
+# Support variants in the future, but this will do for now
+depends_run-append port:clang-9.0
+
+# https://github.com/thepowersgang/mrustc/pull/192
+patchfiles-append patch-mrustc-macos-bootstrap.diff
+
+# https://github.com/thepowersgang/mrustc/pull/199
+patchfiles-append patch-mrustc-llvm-crash.diff
+
+# ld: unknown option: --start-group
+patchfiles-append patch-mrustc-ld.diff
+
+# Add the installed library directory to the library search path
+patchfiles-append patch-mrustc-libdir.diff
+
+# Set the default compiler
+patchfiles-append patch-mrustc-compiler.diff
+
+# Set the default Rust version to match our target
+patchfiles-append patch-mrustc-version.diff
+post-patch {
+    reinplace "s|@MACPORTS_MRUSTC_LIBDIR@|${prefix}/lib/mrustc-${rust_version_major}|" ${worksrcpath}/src/main.cpp
+    reinplace "s|@MACPORTS_MRUSTC_TARGET_VER@|${rust_version_major_underscore}|" ${worksrcpath}/src/main.cpp
+    reinplace "s|@MACPORTS_MRUSTC_CC@|clang-mp-9.0|" ${worksrcpath}/src/trans/codegen_c.cpp
```

Comment:
once you have correctly set the compiler backlisting, please use ${configure.cxx} here instead of hardcoding the compiler name.